manifests/windows/agent6.pp

```diff
@@ -17,21 +17,31 @@
 ) inherits datadog_agent::params {
 
   $msi_full_path = "${msi_location}/datadog-agent-6-${agent_version}.amd64.msi"
+  $msi_source = "${baseurl}ddagent-cli-${agent_version}.msi"
 
   if $ensure == 'present' {
+    if ($agent_version in ['6.14.0', '6.14.1']) {
+        fail("The specified agent version has been blacklisted, please specify a version other than 6.14.0 or 6.14.1")
+    }
+
+    file { 'installer':
```

Comment:
The previous behavior was to only download the MSI if it wasn't already installed. Now we are downloading it regardless. Any reason to prefer this?

Reply:
Maybe I missed something with my testing, but it didn't seem to download regardless. Once the file is in the temp directory I don't think the download was reattempted (ie. this resource doesn't behave like the chef counterpart that always pulls).

That said, we can go back to the original code. I made this change because I wanted to use the `validate_cmd`, but it didn't work particularly well in my tests, so I eventually did the validation in the `exec` resource below. We could still do it the way it was before, but I left it like this because the behavior seemed similar. I can try to reconfirm this doesn't **always** download.

The one thing that sucks, and I might change in a separate PR, is the fact we keep multiple files lying around, using up more and more disk space in temp over time. It might make sense to make `msi_full_path` a constant where we always overwrite with what we pull from the source (if it's not chatty).

Reply:
It doesn't always download, but clearing the temp directory will trigger a download that is not needed if the package is already installed.

Reply:
Tentative fix, yet to be tested: https://github.com/DataDog/puppet-datadog-agent/pull/582